internal/sops/pgp/keysource.go

```diff
@@ -295,7 +295,12 @@ func (key *MasterKey) loadRing(path string) (openpgp.EntityList, error) {
 	if err != nil {
 		return openpgp.EntityList{}, err
 	}
-	defer f.Close()
+	defer func() {
+		cerr := f.Close()
+		if err == nil {
+			err = cerr
```

Comment:
`err` will never be returned on this specific implementation. Note that on the example in the blog you referred to, the `func` using this has a [named return parameter](https://www.geeksforgeeks.org/named-return-parameters-in-golang/) which they leverage to return the error value back to its callers.

Also note that the problem is only when `f.Close()` is called in write operations, which is not the case here as `ReadKeyRing` expects a `io.Reader`.

Reply:
Thanks for the explanation. I thought it would get caught and return by this as `err` is not scoped by `func()` right?
```
if err != nil {
		return keyring, err
```

Reply:
No worries. It would not be caught there as the defer would be the last thing to be executed before returning, so no code executes after it - with the exception of other defers.

Reply:
That makes sense. Thanks for bearing with me as I am new to Golang 😅